tests/test_cacheutils.py

```diff
@@ -167,6 +167,31 @@ def test_lru_basic():
     assert second_lru != lru
 
 
+def test_lru_dict_replacement():
+    # see issue #348
+    cache = LRU()
```

Comment:
I think we can use the same test for both `LRU` and `LRI`.

```suggestion
@pytest.mark.parametrize("lru_class", [LRU, LRI])
def test_lru_dict_replacement(lru_class):
    # see issue #348
    cache = lru_class()
```